### ISSUE TITLE
leveraging parallel environments for sampling faster

### ISSUE THREAD

**Issue by a-z-e-r-i-l-a** [ikostrikov/pytorch-a2c-ppo-acktr-gail#242]
It seems to me that in your implementation usage of parallel environments is not taken advantage from very much.
is not exactly leveraging the usage of parallel environments

Please correct me if I'm wrong. 
My understanding is that one hyperparameter can be the number of samples we obtain before each doing agent update. assuming we know the optimal value of this hyperparameter, using parallel environments should let us gather these samples faster.

If for example we need 4096 samples before each agent update,  it seems to me that your implementation gathers 4096*num_processes before each update, and I wasn't sure if this necessarily boosts the learning speed.
https://github.com/ikostrikov/pytorch-a2c-ppo-acktr-gail/blob/84a7582477fb0d5c82ad6d850fe476829dddd2e1/main.py#L113

one workaround could be for example changing the `for step in range(args.num_steps):` to a while loop that checks the total number of samples that is gathered from all environments.

nevertheless, I would appreciate hearing your view about this :)
